Mage.Sets/src/mage/cards/d/DesertWarfare.java

```diff
@@ -0,0 +1,153 @@
+        this.addAbility(new ConditionalInterveningIfTriggeredAbility(
+                new BeginningOfCombatTriggeredAbility(new DesertWarfareTokenEffect()),
+                new PermanentsOnTheBattlefieldCondition(filter, ComparisonType.OR_GREATER, 5),
+                "At the beginning of combat on your turn, if you control five or more Deserts, "
+                        + "create that many 1/1 red, green, and white Sand Warrior creature tokens. They gain haste."
+        ).addHint(new ValueHint("Deserts you control", new PermanentsOnBattlefieldCount(filter))));
+    }
+
+    private DesertWarfare(final DesertWarfare card) {
+        super(card);
+    }
+
+    @Override
+    public DesertWarfare copy() {
+        return new DesertWarfare(this);
+    }
+}
+
+class DesertWarfareTriggeredAbility extends TriggeredAbilityImpl {
+
+    DesertWarfareTriggeredAbility() {
+        super(Zone.BATTLEFIELD, null, false);
+        setTriggerPhrase("Whenever you sacrifice a Desert and whenever a Desert card is put into your graveyard from your hand or library, "
+                + "put that card onto the battlefield under your control at the beginning of your next end step.");
+    }
+
+    private DesertWarfareTriggeredAbility(final DesertWarfareTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public DesertWarfareTriggeredAbility copy() {
+        return new DesertWarfareTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.SACRIFICED_PERMANENT
+                || event.getType() == GameEvent.EventType.ZONE_CHANGE;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (!(event instanceof ZoneChangeEvent) && !(event instanceof SacrificedPermanentEvent)) {
+            return false;
+        }
+        Effect effect = new ReturnToBattlefieldUnderYourControlTargetEffect();
+        if (event instanceof ZoneChangeEvent) {
+            ZoneChangeEvent zce = (ZoneChangeEvent) event;
+            Card card = game.getCard(event.getTargetId());
+            if (card == null
+                    || !card.isOwnedBy(getControllerId())
+                    || !card.hasSubtype(SubType.DESERT, game)
+                    || zce.getToZone() != Zone.GRAVEYARD
+                    || (zce.getFromZone() != Zone.HAND
+                    && zce.getFromZone() != Zone.LIBRARY)) {
+                return false;
+            }
+            effect.setTargetPointer(new FixedTarget(card, game));
+        } else {
+            Permanent permanent = game.getPermanentOrLKIBattlefield(event.getTargetId());
+            if (permanent == null
+                    || !permanent.isOwnedBy(getControllerId())
+                    || !permanent.hasSubtype(SubType.DESERT, game)) {
+                return false;
+            }
+            effect.setTargetPointer(new FixedTarget(permanent, game));
```

Comment:
Do not use permanent cause it's lki and outdated -- use id instead like `new FixedTarget(event.getTargetId(), game)`